routing/router.go

```diff
@@ -1737,7 +1745,13 @@ func (r *ChannelRouter) SendToRoute(hash lntypes.Hash, rt *route.Route) (
 	}
 
 	err := r.cfg.Control.InitPayment(hash, info)
-	if err != nil {
+	switch {
+	// If this is an MPP attempt and the hash is already registered with
+	// the database, we can go on to launch the shard.
+	case err == channeldb.ErrPaymentInFlight && mpp != nil:
```

Comment:
What if the in-flight payment is not an mpp payment?

Reply:
That's a good point, added DB level check here as well: https://github.com/lightningnetwork/lnd/pull/3970/commits/76e6cc2d5fcbc774adf00a46d9b4cabcfd17325f